default.nix

```diff
@@ -11,6 +11,13 @@ let
   });
   socket-io-src = pkgs.fetchgit (removeAttrs (lib.importJSON ./pkgs/engine-io.json) ["date"]);
   cardano-sl-src = pkgs.fetchgit (removeAttrs (lib.importJSON ./pkgs/cardano-sl.json) ["date"]);
+  cores   = 4;
+  threads = 8;
+  buildSubset = subset: drv: overrideCabal drv (drv: {
+    buildTarget = subset;
+  });
+  buildSpeedupFlagsMarlowScaling = drv : appendConfigureFlag drv
+    "--ghc-option=-rtsopts --ghc-option=+RTS --ghc-option=-N${toString threads} --ghc-option=-A128m --ghc-option=-n2m --ghc-option=-qb0 --ghc-option=-qn${toString cores} --ghc-option=-RTS";
```

Comment:
In fact, @domenkozar, I can't find where `cabal` sets `-N` -- at least it doesn't in https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/haskell-modules/generic-builder.nix

Cabal _does_ set `-j`, but that's `ghc --make`-level parallelism.